lib/lumberjack/server.rb

```diff
@@ -36,49 +39,92 @@ def initialize(options={})
         end
       end
 
-      tcp_server = TCPServer.new(@options[:address], @options[:port])
+      @server = TCPServer.new(@options[:address], @options[:port])
+
+      @close = Concurrent::AtomicBoolean.new
 
       # Query the port in case the port number is '0'
       # TCPServer#addr == [ address_family, port, address, address ]
-      @port = tcp_server.addr[1]
+      @port = @server.addr[1]
 
-      if !@options[:ssl]
-        @server = tcp_server
-      else
+      if @options[:ssl]
         # load SSL certificate
-        ssl = OpenSSL::SSL::SSLContext.new
-        ssl.cert = OpenSSL::X509::Certificate.new(File.read(@options[:ssl_certificate]))
-        ssl.key = OpenSSL::PKey::RSA.new(File.read(@options[:ssl_key]),
+        @ssl = OpenSSL::SSL::SSLContext.new
+        @ssl.timeout = SOCKET_TIMEOUT
```

Comment:
Not sure this is what we want. 'timeout' is for maximum session lifetime. I don't think we should set this?